x-pack/legacy/plugins/code/server/lsp/abstract_launcher.ts

```diff
@@ -161,6 +154,32 @@ export abstract class AbstractLauncher implements ILanguageServerLauncher {
     }
   }
 
+  private async doSpawnProcess(
+    installationPath: string,
+    port: number,
+    log: Logger
+  ): Promise<ChildProcess> {
+    const child = await this.spawnProcess(installationPath, port, log);
+    const pid = child.pid;
+    this.currentPid = pid;
+    this.log.debug('spawned a child process ' + pid);
+    this.spawnTimes += 1;
+    this.startTime = Date.now();
+    this.running = true;
+    if (this.options.lsp.oomScoreAdj && process.platform === 'linux') {
+      try {
+        fs.writeFileSync(`/proc/${pid}/oom_score_adj`, `${OOM_SCORE_ADJ}`);
```

Comment:
@jbudz do you have any idea what the impact of a change like this is? I feel like we definitely shouldn't be writing to `/proc` because we can barely maintain enough permissions to write to the files we actually install.

Reply:
are you worry about we can't write to the file or in general Kibana shouldn't do this level of operation?

Reply:
yeah i'm not sure this would work?  it looks like most of these files are owned by root with 400 permissions

Reply:
for permission, the /proc/$PID should own by the user that starts the process

you could verify by doing this in your shell

`ls -al /proc/$$/oom_score_adj`